lib/workers/repository/init/merge.ts

```diff
@@ -278,3 +281,37 @@ export async function mergeRenovateConfig(
   }
   return returnConfig;
 }
+
+// needed when using portal secrets for npmToken
+export function setNpmTokenInNpmrc(config: RenovateConfig): void {
+  if (!is.string(config.npmToken)) {
+    return;
+  }
+
+  const token = config.npmToken;
+
+  if (!is.string(config.npmrc)) {
+    logger.debug('Adding npmrc to config');
+    config.npmrc = `//registry.npmjs.org/:_authToken=${token}\n`;
```

Comment:
why scoping to registry here and not below?

Reply:
Using the [existing logic](https://github.com/renovatebot/renovate/blob/1562b303449c68e6529ad66a6924db5bf8a36e2a/lib/config/decrypt.ts#L192) in `decryptConfig`

Reply:
@rarkins do you remember why we've two different logics? in this case we explicitly add the default registry and below we don't use it.

Reply:
I'm not sure I understand the question. IIRC the logic is like this:
- If no npmrc, assume npmToken is for the public registry (create a .npmrc)
- If npmrc, and there's a placeholder for NPM_TOKEN already, then substitute it
- If npm, and no placeholder, append it (with no scope?)

Reply:
Yes, that's the current logic, but i don't understand why we assume different scope for point 1 and 3.

So if no `npmrc`we scope to public registry but if there is a `npmrc` config and no `placeholder then we don't scope to public registry. That's confusing behavior.

Reply:
>  i don't understand why we assume different scope for point 1 and 3.

My understanding, In 1 we do not have the npmrc string so we create the full string, whereas in 3 we have a npmrc string present, most possibly with the scope hence we append the auth token to the end.

Reply:
but why scope on point 1 then?  it doesn't need to be scoped then too.

Reply:
The code was added in 2018, maybe it was okay back then. Currently, any docs I can find it does mention to scope the token to the registry rather than adding it in a new line.

Reply:
@rarkins I think the best is to always scope to default registry. If a user needs something different, then they should use the `{NPM_TOKEN}` replacement option instead.

It's probably a breaking behavior, doing in separate PR targeting next major?

Reply:
Sure, that's fine

Reply:
so @RahulGautamSingh leave as is and fix in a follow up pr for next major, which hopefully comes this month